sycl/test/esimd/on-device/PrefixSum.cpp

```diff
@@ -0,0 +1,470 @@
+            cmk_acum_final(buf, it.get_id(0), elem_stride, n_entries);
+          });
+    });
+    return;
+  }
+
+  std::cout << "*** n_entries: " << n_entries << " elem_stide: " << elem_stride
+            << " thread_stride: " << thrd_stride
+            << " entry per thread: " << entry_per_th << std::endl;
+
+  if (entry_per_th == PREFIX_ENTRIES) {
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Accum_iterative1>(
+          range<2>{n_entries / entry_per_th, 1} * range<2>{1, 1},
+          [=](item<2> it) SYCL_ESIMD_KERNEL {
+            cmk_acum_iterative(buf, it.get_id(0), elem_stride, thrd_stride,
+                               PREFIX_ENTRIES);
+          });
+    });
+  } else {
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Accum_iterative2>(
+          range<2>{n_entries / entry_per_th, 1} * range<2>{1, 1},
+          [=](item<2> it) SYCL_ESIMD_KERNEL {
+            cmk_acum_iterative(buf, it.get_id(0), elem_stride, thrd_stride,
+                               PREFIX_ENTRIES_LOW);
+          });
+    });
+  }
+
+  // if number of remaining entries <= 4K , each thread  accumulates smaller
+  // number of entries to keep EUs saturated
+  if (n_entries / entry_per_th > 4096)
+    hierarchical_prefix(q, buf, thrd_stride, thrd_stride * PREFIX_ENTRIES,
+                        n_entries / entry_per_th, PREFIX_ENTRIES);
+  else
+    hierarchical_prefix(q, buf, thrd_stride, thrd_stride * PREFIX_ENTRIES_LOW,
+                        n_entries / entry_per_th, PREFIX_ENTRIES_LOW);
+
+  std::cout << "=== n_entries: " << n_entries << " elem_stide: " << elem_stride
+            << " thread_stride: " << thrd_stride
+            << " entry per thread: " << entry_per_th << std::endl;
+
+  if (entry_per_th == PREFIX_ENTRIES) {
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Prefix_iterative1>(
+          range<2>{n_entries / entry_per_th, 1} * range<2>{1, 1},
+          [=](item<2> it) SYCL_ESIMD_KERNEL {
+            cmk_prefix_iterative(buf, it.get_id(0), elem_stride, thrd_stride,
+                                 PREFIX_ENTRIES);
+          });
+    });
+  } else {
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Prefix_iterative2>(
+          range<2>{n_entries / entry_per_th, 1} * range<2>{1, 1},
+          [=](item<2> it) SYCL_ESIMD_KERNEL {
+            cmk_prefix_iterative(buf, it.get_id(0), elem_stride, thrd_stride,
+                                 PREFIX_ENTRIES_LOW);
+          });
+    });
+  }
+}
+
+//************************************
+// Demonstrate summation of arrays both in scalar on CPU and parallel on device
+//************************************
+int main(int argc, char *argv[]) {
+
+  unsigned int *pInputs;
+#if 0
+    if (argc < 2) {
+        std::cout << "Usage: prefix [N]. N is 2^N entries x TUPLE_SZ" << std::endl;
+        exit(1);
+    }
+    unsigned log2_element = atoi(argv[1]);
+#endif
```

Comment:
Please remove these as it makes hard to read the code.
Though, passing arguments to test binary is nice way to improve its debuggability.

Reply:
Changed to use input argument.